### PR TITLE
non-int indexable base for int()

### PR DESCRIPTION
**Pull request by youknowone**

### CODE REVIEW

File: vm/src/obj/objint.rs [hunk 1]
```diff
@@ -673,11 +673,16 @@ impl IntOptions {
                         "int() can't convert non-string with explicit base".to_owned(),
                     ));
                 }
-                base
+                if let Some(i) = base.payload_if_subclass::<PyInt>(vm) {
+                    i.as_bigint().clone()
+                } else {
+                    let base = vm.call_method(&base, "__index__", vec![])?;
+                    get_value(&base).clone()
+                }
```
Comment (coolreader18) [inline vm/src/obj/objint.rs]: Maybe this could be a function on `vm`? I think it's used in other places as well, at least it is in `list.__getitem__`.

Comment (youknowone) [reply vm/src/obj/objint.rs]: I created a small function but not sure what's the ideal form. Probably adding slot function of this magic method will be helpful to make things clear in future.

Comment (coolreader18) [reply vm/src/obj/objint.rs]: Oh, I was thinking that you could include the `payload_if_subclass::<PyInt>` inside the VM method, as a fast path

Comment (youknowone) [reply vm/src/obj/objint.rs]: Please check if this is ok. Any error message handling suggestion better than Option<PyResult> will be apprecidated.

Comment (coolreader18) [reply vm/src/obj/objint.rs]: Looks good!

**Pull request merged**